### PR TITLE
feat(bridge-ui-v2): Fixed input validations and catching additional errors

### PR DESCRIPTION
**Pull request by KorbinianK**
Bridging ERC20 Tokens was not allowing to send all tokens, as it calculated the fees as if we were paying with the token

Also reordered some validations to stop other validations if we already hit an "error"

### CODE REVIEW

File: packages/bridge-ui-v2/src/components/Bridge/Bridge.svelte [hunk 1]
```diff
@@ -221,6 +230,14 @@
           // TODO: see contract for all possible errors
           errorToast($t('bridge.errors.send_erc20_error'));
           break;
+        case err instanceof UserRejectedRequestError:
+          // Todo: viem does not seem to detect UserRejectError
+          warningToast($t('bridge.errors.rejected'));
+          break;
+        case err instanceof TransactionExecutionError && err.shortMessage === 'User rejected the request.':
```
Comment (jscriptcoder) [inline packages/bridge-ui-v2/src/components/Bridge/Bridge.svelte:237]: Not quite understand why this is needed?

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte [hunk 2]
```diff
@@ -33,17 +37,19 @@
   let inputBox: InputBox;
   let computingMaxAmount = false;
 
+  onMount(() => {
```
Comment (jscriptcoder) [inline packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:40]: Not sure why this is needed

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte [hunk 3]
```diff
@@ -33,17 +37,19 @@
...
+    clearAmount();
```
Comment (jscriptcoder) [inline packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:41]: Why do we want to clear on mount? If the reason is to clear the state, before we actually wanted to leave the amount on purpose when switching pages

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte [hunk 4]
```diff
@@ -33,17 +37,19 @@
...
+
   // Public API
   export function clearAmount() {
     inputBox.clear();
     $enteredAmount = BigInt(0);
   }
 
   export async function validateAmount(token = $selectedToken, fee = $processingFee) {
-    $insufficientBalance = false;
-    $insufficientAllowance = false;
```
Comment (jscriptcoder) [inline packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:44]: Is this fixing a problem? We're gonna validate these two, we start assuming there is no issue and then validate

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte [hunk 5]
```diff
@@ -53,8 +59,38 @@
...
+    const tokenAddress = await getAddress({
+      token,
+      srcChainId: $network.id,
+      destChainId: $destNetwork.id,
+    });
+
+    if (!tokenAddress) return;
+
+    // Check for allowance
+    $insufficientAllowance = await erc20Bridge.requireAllowance({
```
Comment (jscriptcoder) [inline packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:86]: Why do we need to do this here?, we already get the error when running the estimation, don't we?, is this not redundant?

**Review by jscriptcoder — APPROVED** [review]

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte (hunk 3 above)
Comment (KorbinianK) [reply packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:41]: It is not persisting (visibly) right now. We either clear it here or we make sure it is also displayed.

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte (hunk 4 above)
Comment (KorbinianK) [reply packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:44]: yes, it helps with the flickering of the buttons. In the future we could introduce a loading or disabled status

File: packages/bridge-ui-v2/src/components/Bridge/Amount.svelte (hunk 5 above)
Comment (KorbinianK) [reply packages/bridge-ui-v2/src/components/Bridge/Amount.svelte:86]: As discussed in the call, the gas estimation does not throw an error, so we need to call it.

**Review by dantaik — APPROVED** [review]

**Pull request merged**